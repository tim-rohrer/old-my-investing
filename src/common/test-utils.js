// test-utils.js
import React from "react";
// import { render as rtlRender } from '@testing-library/react';
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { createStore } from "redux";
import { Provider } from "react-redux";
import {
  initialStateOriginal as reducerInitialState,
  reducer,
  initialStateOriginal,
} from "../app/store";

// const store = configureStore({ reducer });

export function renderWithRedux(
  ui,
  {
    initialState = initialStateOriginal,
    store = configureStore({ reducer }),
  } = {}
) {
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    // adding `store` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    store,
  };
}
// export function customRender(
//   ui,
//   {
//     initialState: RootState = reducerInitialState,
//     rtlStore,
//     ...renderOptions
//   } = {}
// ) {
//   function Wrapper({ children }) {
//     const rtlStore = configureStore({ reducer });
//     return <Provider store={rtlStore}>{children}</Provider>;
//   }
//   return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
// }

// re-export everything
export * from "@testing-library/react";

// override render method
// export { customRender as render };
