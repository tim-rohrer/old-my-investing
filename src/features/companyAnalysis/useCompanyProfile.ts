import React from "react";
import { useDispatch, useSelector } from "react-redux";
import store, { AppDispatch } from "../../app/store";
import { FMPCompanySymbol } from "../securities/securitiesSlice";
// import store from "../../app/store";
import {
  fetchFMPCompanyProfileBySymbol,
  FMPRequestObject,
  selectCompanyProfile,
  selectProfilesLoaded,
} from "./companyProfilesSlice";
import {
  convertFMPProfileToMyCompanyProfile,
  FMPCompanyProfile,
} from "./fmpUtilities";

export interface MyCompanyProfile {
  symbol: string;
  name: string;
  sector: string;
  industry: string;
  exchange: string;
  description: string;
  ceo: string;
  website: string;
}

type LocalFMPCompanyProfile = FMPCompanySymbol | undefined;

const useCompanyProfile = (symbol: string) => {
  const [companyProfile, setCompanyProfile] = React.useState<
    MyCompanyProfile | undefined
  >();

  const dispatch: AppDispatch = useDispatch();

  const companyProfilesLoaded = useSelector(selectProfilesLoaded);
  const fmpProfile = useSelector(selectCompanyProfile(symbol));

  React.useEffect(() => {
    if (
      companyProfilesLoaded !== null &&
      companyProfilesLoaded.includes(symbol) &&
      fmpProfile !== undefined
    ) {
      setCompanyProfile(() =>
        convertFMPProfileToMyCompanyProfile(symbol, fmpProfile)
      );
    } else {
      dispatch(
        fetchFMPCompanyProfileBySymbol({
          requestType: "companyProfile",
          securitySymbol: symbol,
        })
      );
    }
  }, [companyProfilesLoaded, dispatch, fmpProfile, symbol]);

  return companyProfile;
};

export default useCompanyProfile;
