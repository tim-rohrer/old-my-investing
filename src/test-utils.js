// test-utils.js
import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
// import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { reducer } from './app/store'
// import { initialState as reducerInitialState, reducer } from './reducer'
const reducerInitialState = {}

function render(
  ui,
  {
    initialState: RootState = reducerInitialState,
    rtlStore,
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const rtlStore = configureStore({ reducer })
    return <Provider store={rtlStore}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'

// override render method
export { render }