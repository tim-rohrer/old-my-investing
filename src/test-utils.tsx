// test-utils.tsx
import * as React from "react"
import { Provider } from "react-redux"
import { createStore } from "redux"

import { queries, Queries } from "@testing-library/dom"
import { render as rtlRender, RenderOptions, RenderResult } from "@testing-library/react"

import { rootReducer, State, Store } from "./app/store"

type ExtraOptions = {
  initialState?: State;
  store?: Store;
};

const render = <
  Q extends Queries = typeof queries,
  C extends Element | DocumentFragment = HTMLElement
>(
  ui: React.ReactElement,
  options: ExtraOptions & RenderOptions<Q, C> = {} //:
): RenderResult<Q, C> & { store: Store } => {
  const {
    initialState = undefined, // reducerInitialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = options;

  type Props = {
    children?: React.ReactNode
  }

  const Wrapper = ({ children }: Props) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return {
    ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }),
    store,
  };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
