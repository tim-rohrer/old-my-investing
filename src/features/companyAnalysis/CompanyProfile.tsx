// import { makeStyles } from "@material-ui/core";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { AppDispatch } from "../../app/store";
import { InputWithLabel } from "../../common/InputWithLabel";
import {
  fetchFMPCompanyProfileBySymbol,
  FMPRequestObject,
  selectCompanyProfileError,
} from "./companyProfilesSlice";

// const useStyles = makeStyles((theme) => ({
//   root: {
//     "& .MuiTextField-root": {
//       margin: theme.spacing(1),
//       width: 200,
//     },
//   },
// }));

export const CompanyProfile = () => {
  const [ticker, setTicker] = React.useState("");

  const dispatch: AppDispatch = useDispatch();

  const companyProfileError = useSelector(selectCompanyProfileError);

  const handleSubmit = (event: any) => {
    event.preventDefault();
    setTicker(ticker.toUpperCase());
    const requestObject: FMPRequestObject = {
      requestType: "companyProfile",
      securitySymbol: ticker,
    };
    dispatch(fetchFMPCompanyProfileBySymbol(requestObject));
  };

  const handleChange = (event: {
    target: { value: React.SetStateAction<string> };
  }) => {
    setTicker(event.target.value);
  };

  return (
    <div className="CompanyProfile">
      <header className="CompanyProfile-header">
        <h1>Company Profile</h1>
        {companyProfileError !== undefined ? companyProfileError : null}
      </header>
    </div>
  );
};
