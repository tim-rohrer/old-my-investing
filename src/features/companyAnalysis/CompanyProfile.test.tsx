import React from "react";
import { render, cleanup, screen } from "../../test-utils";
import { CompanyProfile } from "./CompanyProfile";
import { MyCompanyProfile } from "./useCompanyProfile";

const testCompanyProfile: MyCompanyProfile = {
  symbol: "XYZ",
  name: "Triple Letter",
  sector: "Entertainment",
  industry: "Media",
  exchange: "Private",
  description: "A made up company for testing",
  ceo: "Don Juan",
  website: "",
};

afterEach(cleanup);

describe("Company Profile", () => {
  it("renders a snapshot", () => {
    const { container } = render(<CompanyProfile {...testCompanyProfile} />);
    expect(container).toMatchSnapshot();
  });

  it("displays the company name", () => {
    render(<CompanyProfile {...testCompanyProfile} />);
    expect(screen.getByText("Triple Letter")).toBeInTheDocument();
  });
});
