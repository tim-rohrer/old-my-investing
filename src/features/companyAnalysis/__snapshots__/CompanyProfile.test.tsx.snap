// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Profile renders a snapshot 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiPaper-root makeStyles-paper-9 MuiPaper-elevation3 MuiPaper-rounded"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          <h1>
            Triple Letter
          </h1>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
        >
          Entertainment
          /
          Media
          /
          Private
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
          >
            XYZ
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-companyDescription-10 MuiGrid-item MuiGrid-zeroMinWidth MuiGrid-grid-xs-true"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            A made up company for testing
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
